packages/storefront-lwc/src/modules/commerce/productDetail/productDetail.js

```diff
@@ -59,9 +59,12 @@ export default class ProductDetail extends LightningElement {
             if (response.error) {
                 console.error('Error loading product', response.error);
             } else {
+                if (!response.loading) {
+                    this.activeImage = 0;
```

Comment:
This fixes another bug. Imagine the scenario when:

1. The user moves to the 4th image.
2. The user clicks a color variation (that only has 2 images)
3. Now the UI is trying to render 4th index, which doesn't exist

The fix resets the `activeImage` to zero when the back-end provides a new product model (happens for each variation change)